packages/driver/src/cypress/runner.ts

```diff
@@ -492,7 +492,57 @@ const hasOnly = (suite) => {
   )
 }
 
-const normalizeAll = (suite, initialTests = {}, setTestsById, setTests, getRunnableId, getHookId, getOnlyTestId, getOnlySuiteId, createEmptyOnlyTest) => {
+const pruneSuite = (emptySuite) => {
+  emptySuite.cleanReferences()
+
+  if (emptySuite.parent) {
+    const parentSuites = emptySuite.parent.suites.filter((suite) => suite !== emptySuite)
+
+    emptySuite.parent.suites = parentSuites
+    emptySuite.parent._onlySuites = emptySuite.parent._onlySuites.filter((suite) => parentSuites.includes(suite))
+    emptySuite.parent = null
+  }
+}
+
+const pruneEmptySuites = (rootSuite, incrementFoundTestsBy) => {
+  for (const suite of [...rootSuite.suites]) {
```

Comment:
```suggestion
 if (!Cypress.testFilter) {
 	return // do nothing if the test filter is not set.
 }

  for (const suite of [...rootSuite.suites]) {
```

Reply:
We still want to prune empty suites regardless of a `testFilter` being present or not so that the UI doesn't show the empty suite dropdowns.

Reply:
Revisiting this, can I get your opinion on what you think the desired behavior is regarding empty suites e.g. `describe('empty', () => {})`?

Currently they show up in the UI as an empty runnable. Looks a bit wonky but has no side effects. With the way things are wired up now, these suites are pruned (regardless of if there is a test filter or not). What could be confusing is if a developer goes to test some new functionality and starts with `describe.only('new test coverage', () => {})` and they hit save. Since the suite has no children, it will get pruned and the `only` will not apply. This behavior seems pretty confusing to me

Reply:
I moved this to an earlier check, we won't call `pruneEmptySuites` unless there is a filter set.

Reply:
I think this is a good start on this behavior. If we prune empty suites, we will end up with different run results in `run` mode which will remove the reporting of these empty suites to the dashboard. Even though there is nothing to "run" it does help visually indicate that a test has been stubbed out.

I do think it's confusing we have differences between empty suites vs empty test behaviors in the reporter visuals, but seems like a small fix in the reporter to show those consistently if we wanted to go that route.